### PR TITLE
anthropic[patch]: correctly handle tool msg with empty list

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/partners/anthropic/langchain_anthropic/chat_models.py [hunk 1]
```diff
@@ -118,9 +118,13 @@ def _merge_messages(
     for curr in messages:
         curr = curr.model_copy(deep=True)
         if isinstance(curr, ToolMessage):
-            if isinstance(curr.content, list) and all(
-                isinstance(block, dict) and block.get("type") == "tool_result"
-                for block in curr.content
+            if (
+                isinstance(curr.content, list)
+                and curr.content
```
Comment (efriis) [inline libs/partners/anthropic/langchain_anthropic/chat_models.py:123]: nit: maybe worth adding a code comment for posterity

**Review by efriis — APPROVED** [review]
looks good. should we add testing for this to standard tests instead? seems like an edge case we should test across all providers
